### PR TITLE
feat: history page

### LINKED ISSUE #18: History page

### PR DESCRIPTION
**Pull request by basecode**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #18

Test URLs:
- Before: https://main--mammotome--hlxsites.hlx.page/en/about-us/our-history
- After: https://18-history-page--mammotome--hlxsites.hlx.page/en/about-us/our-history

### CODE REVIEW

File: styles/styles.css [hunk 1]
```diff
@@ -314,3 +314,82 @@ main .section[data-section-status='loading'],
...
+  font-size: 1rem;
+  font-weight: 900;
+  text-transform: uppercase;
+  letter-spacing: 4px;
+  margin-bottom: 100px;
+}
+
+body.our-history main .year h3 {
+  font-size: var(--mt-font-size-100);
+  color: var(--mt-global-color-white);
```
Comment (solaris007) [inline styles/styles.css:348]: would probably use: `--mt-background-color-primary` which references white

File: styles/styles.css [hunk 2]
```diff
@@ -314,3 +314,82 @@ main .section[data-section-status='loading'],
...
+  color: var(--mt-global-color-base-primary);
+  font-size: 1rem;
+  font-weight: 900;
+  text-transform: uppercase;
+  letter-spacing: 4px;
+  margin-bottom: 100px;
+}
+
+body.our-history main .year h3 {
+  font-size: var(--mt-font-size-100);
```
Comment (solaris007) [inline styles/styles.css:347]: would probably use: `--mt-body-font-size-m` which references 100

File: scripts/scripts.js [hunk 3]
```diff
@@ -57,6 +145,8 @@ export async function decorateMain(main) {
   buildAutoBlocks(main);
   decorateSections(main);
   decorateBlocks(main);
+  decorateHistoryPage(main);
```
[the remark this answers is not in the document]
Comment (basecode) [reply scripts/scripts.js]: Moved everything into a section.

**Comment by solaris007** [comment]
behavioral differences:

desktop:
- item scrolling out of view at the top becomes blue again, should stay purple
- item scrolling into view from bottom is already purple, but should be blue
- top part of the line stays blue instead of purple
- bottom part of the line is purple, but should be blue from screen center
- horizontal distance between items/year and line is larger than on original
- font and/or padding in items is different, as evidenced by different line wrapping and slightly larger box sizes

some of the above also applies to mobile.

Screenshot 2023-03-20 at 07 36 02
Screenshot 2023-03-20 at 07 30 21

File: styles/styles.css [hunk 4]
```diff
@@ -306,3 +314,150 @@ main .section[data-section-status='loading'],
...
+main .section.our-history .year h3 {
+  font-size: var(--mt-body-font-size-100);
+  color: var(--mt-global-color-white);
+}
+
+main .section.our-history .year::before {
+  display: inline-block;
+  content: '';
+  background-color: var(--mt-our-history-color-blue);
+  background-image: url('/icons/calendar.svg');
```
Comment (solaris007) [inline styles/styles.css:360]: better have the icon as text content `:calendar:` and then use `await decorateIcons(yourSection);`

this way, the icon sprite is leveraged.

find example usages in header.js/footer.js.

File: scripts/scripts.js [hunk 5]
```diff
@@ -57,6 +62,11 @@ export async function decorateMain(main) {
   buildAutoBlocks(main);
   decorateSections(main);
   decorateBlocks(main);
+
+  if (main.querySelector('.section.our-history')) {
```
[the remark this answers is not in the document]
Comment (basecode) [reply scripts/scripts.js:66]: would require `buildAutoBlocks` to know the sections, hence re-ordering of function execution. we decided to not do it.

**Review by solaris007 — APPROVED** [review]

**Pull request merged**